index.js

```diff
@@ -243,31 +243,59 @@ MediaSession.prototype = extend(MediaSession.prototype, {
             self.constraints = renegotiate;
         }
 
-        this.pc.handleOffer({
-            type: 'offer',
-            jingle: this.pc.remoteDescription
-        }, function (err) {
-            if (err) {
-                self._log('error', 'Could not create offer for adding new stream');
-                return cb(err);
-            }
-            self.pc.answer(self.constraints, function (err, answer) {
+        if (this.pc.isInitiator) {
+            this.pc.offer(self.constraints, function (err, offer) {
                 if (err) {
-                    self._log('error', 'Could not create answer for adding new stream');
+                    self._log('error', 'Could not create offer for adding new stream');
                     return cb(err);
                 }
-                answer.jingle.contents.forEach(function (content) {
-                    filterContentSources(content, stream);
-                });
-                answer.jingle.contents = answer.jingle.contents.filter(function (content) {
-                    return content.description.descType === 'rtp' && content.description.sources && content.description.sources.length;
+                self.pc.handleAnswer({
+                    type: 'answer',
+                    jingle: self.pc.remoteDescription
+                }, function (err) {
```

Comment:
Yes, when you change an offer, you always have to answer afterward, and when you change an answer, you always have to offer first.